internal/querycoordv2/checkers/leader_checker.go

```diff
@@ -0,0 +1,205 @@
+		meta:              meta,
+		dist:              dist,
+		target:            target,
+		nodeMgr:           nodeMgr,
+	}
+}
+
+func (c *LeaderChecker) ID() utils.CheckerType {
+	return utils.LeaderChecker
+}
+
+func (c *LeaderChecker) Description() string {
+	return "LeaderChecker checks the difference of leader view between dist, and try to correct it"
+}
+
+func (c *LeaderChecker) Check(ctx context.Context) []task.Task {
+	if !c.IsActive() {
+		return nil
+	}
+
+	collectionIDs := c.meta.CollectionManager.GetAll()
+	tasks := make([]task.Task, 0)
+
+	for _, collectionID := range collectionIDs {
+		collection := c.meta.CollectionManager.GetCollection(collectionID)
+		if collection == nil {
+			log.Warn("collection released during check leader", zap.Int64("collection", collectionID))
+			continue
+		}
+
+		replicas := c.meta.ReplicaManager.GetByCollection(collectionID)
+		for _, replica := range replicas {
+			for _, node := range replica.GetNodes() {
+				if ok, _ := c.nodeMgr.IsStoppingNode(node); ok {
+					// no need to correct leader's view which is loaded on stopping node
+					continue
+				}
+
+				leaderViews := c.dist.LeaderViewManager.GetByCollectionAndNode(replica.GetCollectionID(), node)
+				for ch, leaderView := range leaderViews {
+					dist := c.dist.SegmentDistManager.GetByShardWithReplica(ch, replica)
+					tasks = append(tasks, c.findNeedLoadedSegments(ctx, replica.ID, leaderView, dist)...)
+					tasks = append(tasks, c.findNeedRemovedSegments(ctx, replica.ID, leaderView, dist)...)
+				}
+			}
+		}
+	}
+
+	return tasks
+}
+
+func (c *LeaderChecker) findNeedLoadedSegments(ctx context.Context, replica int64, leaderView *meta.LeaderView, dist []*meta.Segment) []task.Task {
+	log := log.Ctx(ctx).With(
+		zap.Int64("collectionID", leaderView.CollectionID),
+		zap.Int64("replica", replica),
+		zap.String("channel", leaderView.Channel),
+		zap.Int64("leaderViewID", leaderView.ID),
+	)
+	ret := make([]task.Task, 0)
+	dist = utils.FindMaxVersionSegments(dist)
+	for _, s := range dist {
+		version, ok := leaderView.Segments[s.GetID()]
+		currentTarget := c.target.GetSealedSegment(s.CollectionID, s.GetID(), meta.CurrentTarget)
+		existInCurrentTarget := currentTarget != nil
+		existInNextTarget := c.target.GetSealedSegment(s.CollectionID, s.GetID(), meta.NextTarget) != nil
+
+		if !existInCurrentTarget && !existInNextTarget {
+			continue
+		}
+
+		leaderWithOldVersion := version.GetVersion() < s.Version
+		// leader has newer version, but the query node which loaded the newer version has been shutdown
+		leaderWithDirtyVersion := version.GetVersion() > s.Version && c.nodeMgr.Get(version.GetNodeID()) == nil
+
+		if !ok || leaderWithOldVersion || leaderWithDirtyVersion {
+			log.Debug("leader checker append a segment to set",
+				zap.Int64("segmentID", s.GetID()),
+				zap.Int64("nodeID", s.Node))
+			action := task.NewSegmentActionWithScope(s.Node, task.ActionTypeGrow, s.GetInsertChannel(), s.GetID(), querypb.DataScope_Historical)
```

Comment:
Suggest to add a new action type as:
- This impl changes the load scope to `Delta` case by case
- QueryNode doesn't impl `remove segment from leader view` as `ReleaseSegment`, but this PR does changed to

Reply:
1. we need to set load scope for now case by case, even a new action type exist. such as the `SegmentUpdateAction`
2. actually query node does remove segment from leader view by `ReleaseSegment`